### PR TITLE
Tickets/dm 38195

### PR DESCRIPTION
**Pull request by gmegh**

### CODE REVIEW

File: python/lsst/ts/externalscripts/auxtel/latiss_base_align.py [hunk 1]
```diff
@@ -287,29 +287,40 @@ async def take_intra_extra(self) -> None:
...
+            Ry offset, in degrees (default=0).
+        m1 : `float`, optional
+            M1 pressure offset, in Pa (default=0).
         """
 
         offset = {
-            "m1": 0.0,
-            "m2": 0.0,
+            "m1": m1,
+            "m2": 0,
```
Comment (tribeiro) [inline python/lsst/ts/externalscripts/auxtel/latiss_base_align.py:318]: This is a float, so probably better to keep it as `0.0`

File: python/lsst/ts/externalscripts/auxtel/latiss_base_align.py [hunk 2]
```diff
@@ -495,6 +506,43 @@ def get_schema(cls) -> typing.Dict[str, typing.Any]:
                     for possible debugging scenarios.
                 type: boolean
                 default: false
+              offset_x:
+                description: x degree of freedom offset in mm.
+                anyOf:
```
Comment (tribeiro) [inline python/lsst/ts/externalscripts/auxtel/latiss_base_align.py:511]: I think you need something like this for all the new parameters instead:

```
type: number
default: 0.0
```

File: python/lsst/ts/externalscripts/auxtel/latiss_base_align.py [hunk 3]
```diff
@@ -495,6 +506,43 @@ def get_schema(cls) -> typing.Dict[str, typing.Any]:
...
+                default: null
+              offset_Ry:
+                description: Ry degree of freedom offset in degrees.
+                anyOf:
+                  - type: integer
+                  - type: "null"
+                default: null
+              offset_m1:
+                description: M1 pressure degree of freedom offset in Pa.
+                anyOf:
```
Comment (tribeiro) [inline python/lsst/ts/externalscripts/auxtel/latiss_base_align.py:541]: This one would be something like:

```
type: number
default: 0.0
maximum: 0.0
```

File: python/lsst/ts/externalscripts/auxtel/latiss_base_align.py [hunk 4]
```diff
@@ -495,6 +506,43 @@ def get_schema(cls) -> typing.Dict[str, typing.Any]:
                     for possible debugging scenarios.
                 type: boolean
                 default: false
+              offset_x:
```
Comment (tribeiro) [inline python/lsst/ts/externalscripts/auxtel/latiss_base_align.py:509]: Oh! these should not be in the `LatissBaseAlign` configuration.

I think having the `additional_configuration` method and having the additional offsets in the other class was better.

**Review by tribeiro — CHANGES_REQUESTED** [review]
Hear some initial comments. I think the most critical thing now is putting back the `additional_configuration` from your previous implementation and having the additional configurations in `LatissIntraExtraFocalData` instead of `LatissBaseAlign`.

File: python/lsst/ts/externalscripts/auxtel/latiss_base_align.py [hunk 5]
```diff
@@ -287,29 +287,40 @@ async def take_intra_extra(self) -> None:
...
+        offset: float,
+        x: float = 0.0,
+        y: float = 0.0,
+        u: float = 0.0,
+        v: float = 0.0,
+        m1: float = 0.0,
     ) -> None:
-        """Applies z-offset to the hexapod to move between
-        intra/extra/in-focus positions.
+        """Applies offset to the hexapod degrees of freedom.
```
Comment (tribeiro) [inline python/lsst/ts/externalscripts/auxtel/latiss_base_align.py:298]: so, `xyzuv` are hexapod but m1 is not... Maybe rephrase it:

```
Applies offsets to the AOS look-up table values.
```

We should probably rename the method as well... Maybe `offset_aos`? If you use VSCode you can easily rename method. Let me know if you need any help.

File: python/lsst/ts/externalscripts/auxtel/latiss_base_align.py [hunk 6]
```diff
@@ -287,29 +287,40 @@ async def take_intra_extra(self) -> None:
...
 
         Parameters
         ----------
         offset : `float`
-             Focus offset to the hexapod in mm.
+            Offset to the hexapod in the z-axis mm.
```
Comment (tribeiro) [inline python/lsst/ts/externalscripts/auxtel/latiss_base_align.py:303]: Remove "to" after "Offset:

```
Offset the hexapod in the z-axis, in mm.
```

File: python/lsst/ts/externalscripts/auxtel/latiss_intra_extra_focal_data.py [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+        # set dofs
+        self.offset_x = config.offset_x
+        self.offset_y = config.offset_y
+        self.offset_z = config.offset_z
+        self.offset_Rx = config.offset_Rx
+        self.offset_Ry = config.offset_Ry
+        self.offset_m1 = config.offset_m1
+
+    async def arun(self, checkpoint: bool = False):
+        """Takes Intr and Extra Focal Data.
```
Comment (tribeiro) [inline python/lsst/ts/externalscripts/auxtel/latiss_intra_extra_focal_data.py:125]:
```
Run script operation encapsulating the checkpoints to allow running standalone.
```

If you want to describe what the method does, maybe add a "Notes" session. But I am not sure this is necessary.

File: python/lsst/ts/externalscripts/auxtel/latiss_intra_extra_focal_data.py [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+            self.offset_y,
+            self.offset_Rx,
+            self.offset_Ry,
+            self.offset_m1,
+        )
+
+        # Setting visit_id's to none so run_cwfs will take a new dataset.
+        self.intra_visit_id = None
+        self.extra_visit_id = None
+        await self.take_intra_extra()
```
Comment (tribeiro) [inline python/lsst/ts/externalscripts/auxtel/latiss_intra_extra_focal_data.py:162]: Just to be in the safe side, maybe add a "try/except" clause here to make sure resetting the offset always happens:

```
try:
    await self.take_intra_extra()
finally:
    await self.hexapod_offset(...)
```

**Review by tribeiro — COMMENTED** [review]
A couple more comments.

**Comment by gmegh** [comment]
All changes made. Regarding the name of the method hexapod_offsets, could we call it dof_offsets? or you don’t like the name dof? I feel "LUT offsets" is not very clear to someone unfamiliar with LUT…

**Comment by tribeiro** [comment]
You need to include:

```
from .latiss_intra_extra_focal_data import *
```

in this [__init__](https://github.com/lsst-ts/ts_externalscripts/blob/develop/python/lsst/ts/externalscripts/auxtel/__init__.py) file.

File: python/lsst/ts/externalscripts/auxtel/latiss_intra_extra_focal_data.py [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+        $id: {url}{path}
+        title: LatissIntraExtraFocalData v1
+        description: Configuration for making a LATISS calibrations SAL Script.
+        type: object
+        properties:
+            offset_x:
+                description: x degree of freedom offset in mm.
+                type: number
+                default: 0.0
+              offset_y:
```
Comment (tribeiro) [inline python/lsst/ts/externalscripts/auxtel/latiss_intra_extra_focal_data.py:78]: Can you make sure the indentation on all these entries match?

File: doc/version_history.rst [hunk 10]
```diff
@@ -6,6 +6,17 @@
 Version History
 ===============
 
+v0.23.5
```
Comment (tribeiro) [inline doc/version_history.rst:9]: I think this probably deserves a minor release! Can you make this v0.24.0, please?

File: doc/version_history.rst [hunk 11]
```diff
@@ -6,6 +6,17 @@
...
+-------
+
+* Add new Script ``LatissIntraExtraFocalData`` for AuxTel.
+  This script is used to take intra and extra focal data with given look up table offsets.
+  It uses the ``latiss_base_align.py`` module.
+
+* In ``latiss_base_algin.py``:
```
Comment (tribeiro) [inline doc/version_history.rst:16]: typo: 
latiss_base_algin.py -> latiss_base_align.py

**Review by tribeiro — APPROVED** [review]
Alright! Looks good! thanks for all the updates and hard work! I left a few final comments in the version history.. Sorry I didn't caught them earlier..
